.github/workflows/style-and-lint.yaml

```diff
@@ -0,0 +1,88 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+# Only run when PR is opened to the main branch, and changes are made to these types of R files
+on:
+  pull_request:
+    paths: ["**.[rR]", "**.[rR]md", "**.[rR]markdown", "**.[rR]nw"]
+    branches: [main]
```

Comment:
I think the syntax is correct as written ([Actions documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches)). But whether we only want it to run on PRs to `main` or just on all PRs is a reasonable question. 

It may not be running yet because the actions don't exist on `main` - so we might need to remove that line anyway if we want to see what happens when the actions run within the scope of this PR prior to merging into `main`. Does that sound reasonable? I'm not 100% sure if that is correct.

Reply:
Yeah, looks like you're right. I'm happy merging this now and then I'll make a PR sometime this afternoon with some style updates to see how the PR report looks.

Reply:
🤣so I said I would merge and then promptly closed the tab....